docs/extend/plugins.md

```diff
@@ -1,87 +1,272 @@
 <!--[metadata]>
 +++
-title = "Extending Engine with plugins"
-description = "How to add additional functionality to Docker with plugins extensions"
-keywords = ["Examples, Usage, plugins, docker, documentation, user guide"]
+aliases = [
+"/engine/extend/"
+]
+title = "New Plugin System"
```

Comment:
Having said that, naming this file `index.md` could be a good idea, then we can:

- redirect /engine/extend/plugins/ to /engine/extend/legacy_plugins/
- use /engine/extend/ for the new content (although I never was fond of "extend", as I think "plugins" is more descriptive)